goldstone/core/models.py

```diff
@@ -2634,3 +2636,70 @@ def integration(cls):
     def resourcetype(cls):
 
         return "routers"
+
+
+class DefinedSearch(models.Model):
+    """Defined searches, both system and user-created."""
+
+    # This object's Goldstone UUID.
+    uuid = UUIDField(version=1, auto=True, primary_key=True)
+
+    name = models.CharField(max_length=64)
```

Comment:
Does it make sense to differentiate owners of searches?  That could simply be core, compliance, and user for now.  In that case I would make the name,owner tuple unique rather than just the name.


Thanks,

John
@jxstanford

> On Dec 10, 2015, at 15:55, John DeRosa <notifications@github.com> wrote:
> 
> In goldstone/core/models.py:
> 
> > @@ -2634,3 +2636,70 @@ def integration(cls):
> >      def resourcetype(cls):
> >  
> >          return "routers"
> > +
> > +
> > +class DefinedSearch(models.Model):
> > +    """Defined searches, both system and user-created."""
> > +
> > +    # This object's Goldstone UUID.
> > +    uuid = UUIDField(version=1, auto=True, primary_key=True)
> > +
> > +    name = models.CharField(max_length=64)
> The name field should be defined to be unique in the model.
> 
> —
> Reply to this email directly or view it on GitHub.
>